p2p/discover/udp.go

```diff
@@ -542,9 +546,12 @@ func (t *udp) handlePacket(from *net.UDPAddr, buf []byte) error {
 	err = packet.handle(t, from, fromID, hash)
 	unixTime := float64(time.Now().UnixNano()) / 1000000000
 	// if NEIGHBORS packet, add the node address info to the sql database
-	if t.sqldb != nil && packet.name() == "RLPX_NEIGHBORS" {
```

Comment:
It looks like you move the error deeper - any particular reason for this? It seems like there is no difference in the output, but in the new version it will iterate through the Nodes pointlessly

Reply:
Yes. Similar to other changes to the db handle checking, I made it so that node-finder continues doing all tasks unrelated to updating the sql db, such as logging messages.
If sql database connection info was provided at run-time, then any sql db handle, `t.sqldb`, `srv.DB`, `pm.db`, will not be nil. If not provided (therefore the value is nil), then the instance was meant to run without connection to sql database and should everything else normally.